Telerik.Sitefinity.Amazon/BlobStorage/AmazonBlobStorageProvider.cs

```diff
@@ -26,25 +26,42 @@ public class AmazonBlobStorageProvider : CloudBlobStorageProvider
         /// <param name="config">The collection of parameters (each by its name and value) of the current provider's configuration settings.</param>
         protected override void InitializeStorage(NameValueCollection config)
         {
+            var useIamInstanceRoleValue = "false";
+            if (config.AllKeys.Contains(UseIamInstanceRoleKey))
+                useIamInstanceRoleValue = config[UseIamInstanceRoleKey].Trim();
+
+            if (!bool.TryParse(useIamInstanceRoleValue, out this.useIamInstanceRole))
+                throw new ConfigurationException("'{0}' unable to parse {1}".Arrange(UseIamInstanceRoleKey, useIamInstanceRoleValue));
+
             this.accessKeyId = config[AccessKeyIdKey].Trim();
-            if (String.IsNullOrEmpty(this.accessKeyId))
+            if (!this.useIamInstanceRole && String.IsNullOrEmpty(this.accessKeyId))
```

Comment:
Change to AccessKeyId and SecretKeyId parameters to only validate if we are not using the EC2 Instance Profile Role